### ISSUE TITLE
Link inside Cell causes page refresh

### ISSUE THREAD

**Issue by arkn98** [tannerlinsley/react-table#1329]
I'm not being rude or anything, and I definitely understand everybody has work to do and it is difficult to spend time on issues. But my question, which in my opinion is valid, on the spectrum forum, which was over a month ago, didn't receive a single comment (I deleted my account after that). And I can't find a better way to solve that without opening an issue here.

I've posted the question again [here](https://spectrum.chat/react-table/general/adding-link-from-react-router-dom-inside-a-cell-causes-a-page-refresh~20c79a4d-bd38-45dd-8fe4-6898f8884819). And I will post it here as well.

A Link (from [react-router-dom](https://reacttraining.com/react-router/web)) inside a Cell causes the entire page to refresh. **But the same link outside the `react-table` works just as it is supposed to (no refresh)**. My `react-table` has no special configuration; just has custom styling. The snippet I used is from one of the [issues](https://github.com/tannerlinsley/react-table/issues/89), which goes like this:

    ...,
    Cell: ({ row }) => (
        <Link to={{ pathname: `/dashboard/list/${row.id}` }}>{row.id}</Link>
    ),
    ...

The link is clickable and goes to the right component. But it causes a refresh. Is this expected behavior or am I going wrong somewhere?

Other useful info:

    "react-table": "^6.9.2",
    "react": "^16.8.4",
    "react-dom": "^16.8.4",
    "react-scripts": "^2.1.8",
    "react-router-dom": "^4.4.0",

**Comment by tannerlinsley** [comment]
I'm sorry to hear about your experience. The triaging flow is meant to cover the majority case, but clearly this is not one of them. I'll do my best to help here.

As far as I know, there is absolutely no logic inside React Table that interacts with browser history or reloads. My first thought of debugging this however would be to pause the debugger inside of the `Link` component and dive into each function until you find a potential `window.location.reload` call or `window.location.href = 'something/new'` or `window.history.push/replace`. If you go through the entire call chain and don't encounter anything, then presumably, Reach Router will resolve the new navigation and render the next component.

This is where things would get tricky. By this point, I would probably set a debug inside of Reach Router's `history.js` file on these lines: https://github.com/reach/router/blob/master/src/lib/history.js#L50-L52. If either of those is firing when you don't want it to, then you'll want to walk up the call stack to see what is firing those and why.

If you *still* can't find the cause, then a last resort would be to do a global search in your chrome dev tools  (option + command + F) and search for anything using `window.location` or `window.history`, especially `window.location.reload` or `window.history.push/replace`.

Bottom line is, if your page is reloading, we can find what it is, it just takes time.

Other weird things that might be happening... maybe a form element is being submitted without the `event.preventDefault()` being called.

Either way. Let's try and figure this out.

**Comment by tannerlinsley** [comment]
Another thing - Have you tried the v7 alpha yet? I would be interested to see if this happens there as well. There is a link to the sandbox you could play around with in the Readme.

**Comment by arkn98** [comment]
Thanks for the quick reply. I'll try what you said and get back. By the way, the package I used is `react-router` not reach router (correcting for people who may refer to this in the future). I'll also try v7 alpha with a similar setting.

**Comment by tannerlinsley** [comment]
React Router changes things a bit, but the concepts are the still the same. Look for the same methods and calls in their source code and debug from there.

**Comment by arkn98** [comment]
Okay. I found out what was the cause. I use `react-click-outside` from [here](https://github.com/tj/react-click-outside/) on several components (like popups) to close them when clicked outside the component. This adds an event listener to the entire document, but propagates the events properly (`Link` components outside the `react-table` work properly even with `react-click-outside`). After I removed that listener, there was no page refresh. Any reason for this behavior to links exclusively inside `react-table`?

**Comment by tannerlinsley** [comment]
There are a few `event.stopPropagation()`s inside of the `methods.js` file, but only for column resizing. Otherwise, there is no event manipulation in React Table. Do you have any wrappers or HOC's around React Table that could be dealing with event propagation?

**Comment by arkn98** [comment]
Yeah. I had a `withRouter` HOC (which is also from `react-router-dom`) for programmatically changing routes within my component. I removed it and the `Link` now works fine. 

Real sorry for wasting your time. Thanks again. Also request you to remove the `bug` tag from this issue.

**Comment by tannerlinsley** [comment]
Glad we got it figured out!

**Comment by jogoool** [comment]
I have a side menu that will close when I clicked any area of the document ( adding addEventListener to the document). now I face a problem with clicking the value of the cell, the onClick is not working with the Cell (custom rendering), even when I use getProps or getTdProps, I tried e.stopPropagation() and e.nativeEvent.stopImmediatePropagation(), unfortunately it was unsuccessful.
in the normal rendering of the column when I clicked the value of cell the onClick event is triggering but when I use custom cell rendering (using Cell ) the onClick event is not triggering.
